src/test/tgen/fixed_size/gen_conf.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+
+import sys
+import networkx as nx
+
+def main():
+    generate_server()
+    generate_client(1, "1 mib", 10, "client.1stream_1mib_10x.graphml")
+    generate_client(10, "1 mib", 10, "client.10streams_1mib_10x.graphml")
+    generate_client(1, "1 kib", 100, "client.1stream_1kib_100x.graphml")
+    generate_client(100, "1 kib", 100, "client.100streams_1kib_100x.graphml")
+    generate_client(1, "1 b", 1000, "client.1stream_1b_1000x.graphml")
+    generate_client(1000, "1 b", 1000, "client.1000streams_1b_1000x.graphml")
+
+def generate_server():
+    G = nx.DiGraph()
+    G.add_node("start", serverport="80", stallout="60 seconds")
+    nx.write_graphml(G, "server.graphml")
+
+def generate_client(num_streams, transfer_size, count, filename):
+    G = nx.DiGraph()
+
+    # We start client and server at the same time, but client waits one second to connect.
+    G.add_node("start", time="1 second", peers="server:80", stallout="60 seconds", sendsize=transfer_size, recvsize=transfer_size)
+    G.add_node("pause")
+    G.add_node("end", count=str(count))
+
+    for i in range(num_streams):
+        stream_name = f"stream{i}"
+        # Bidirectional transfer.
+        G.add_node(stream_name)
+        # Streams are launched in parallel.
+        G.add_edge("start", stream_name)
+        # Waits until all parallel streams are finished before checking end conditions.
+        G.add_edge(stream_name, "pause")
+
+    # If end conditions are not met, start another set of streams.
+    G.add_edge("pause", "end")
+    G.add_edge("end", "start")
```

Comment:
What does this mean? So there are X total streams, and only Y run in parallel? And they are grouped, so it waits for Y parallel streams to complete, then starts another Y streams, until X total streams are complete?

Reply:
Yes, that's correct. The `pause` node without a specified pause time attribute acts as a "barrier" that causes tgen to wait until all incoming edges visits that node before it continues its walk through the graph.

In this case I am setting X to the same as Y so there will only be 1 round of "parallel" streams: the pause means tgen will not move forward until those Y streams complete, and then it moves to the end node which checks that X streams completed and ends because X==Y. But we could set up multiple rounds if we wanted by setting X greater than Y.

I'll add some comments to the script. :)